nautobot/docs/user-guides/git-data-source.md

```diff
@@ -0,0 +1,269 @@
+
+### Config Contexts
+Detailed information on [config contexts](../models/extras/gitrepository.md#configuration-contexts) in Git Repositories.
+
+Config contexts may be provided as JSON or YAML files located in `/config_contexts/`.
+
+Config contexts can be used to provide additional details to different automation tooling.  For example Ansible variables, or any other data that you can't natively store in Nautobot.  It can also be used in the Golden Configuration Nautobot plugin to provide extra details to generate configuration templates.
+
+A few simple examples of Configuration Context data might be:
+ - DNS Servers
+ - NTP Servers
+ - ACL Data
+ - Routing Information such as BGP ASNs etc.
+
+Similar to the other data sources, the repository can be added by navigating to **Extensibility -> Git repositories**. Click on **Add**, and fill out the repository details.
+
+![](./images/git-data-source_13.png)
+
+Once the repository syncs the details can be found in the **Synchronization Status** tab.  For example the platform specifics were synced.
+
+![](./images/git-data-source_14.png)
+
+The repository structure is:
+```
+▶ tree config_contexts 
+config_contexts
+├── devices
+│   ├── site-a-bb-01.yml
+│   ├── site-a-rtr-01.yml
+│   ├── site-a-rtr-02.yml
+│   ├── site-a-spine-01.yml
+│   ├── site-a-spine-02.yml
+│   ├── site-b-bb-01.yml
+│   ├── site-b-leaf-01.yml
+│   ├── site-b-leaf-02.yml
+│   ├── site-b-rtr-01.yml
+│   ├── site-b-rtr-02.yml
+│   ├── site-b-spine-01.yml
+│   └── site-b-spine-02.yml
+├── platform_eos.yml
+├── platform_junos.yml
+├── platform_nxos.yml
+└── role_spine.yml
+
+1 directory, 16 files
+```
+
+Configuration Context details:
+- Follows an inheritance methodology similar to what Ansible implements.  Global contexts can be overwritten by local contexts at both a group level, as well as at a device specific level.
+- Nautobot UI provides a simple view to see merged config contexts.  It can be visualized by navigating to a device and clicking on the **config contexts** tab.
+
+Here's an example, with some of the details omitted for brevity.
+
+![](./images/git-data-source_15.png)
+
+There is a huge benefit to having `config contexts` managed by a Git workflow.  This type of data can be modified often, especially platform specifics, or new device roles.  Utilizing a standard Git workflow allows for all the proper reviews and approvals to be accomplished before accepting the changes into Nautobot for use.
+
+
+## Additional Git Data Sources
+
+As seen in [Step 2](#Step-2), the standard installation of Nautobot will come natively with export templates, jobs, and config contexts.  Additional data sources can be incorporated using the Nautobot plugin system.  For example, the [nautobot-plugin-golden-config](https://github.com/nautobot/nautobot-plugin-golden-config) plugin implements four additional data sources.
```

Comment:
```suggestion
As seen in [Step 2](#step-2), the standard installation of Nautobot will come natively with export templates, jobs, and config contexts.  Additional data sources can be incorporated using the Nautobot plugin system.  For example, the [nautobot-plugin-golden-config](https://github.com/nautobot/nautobot-plugin-golden-config) plugin implements four additional data sources.
```

Though it might be better to retitle "Step 1" through "Step 3" into something more specific and meaningful?